go-controller/pkg/ovn/egressip.go

```diff
@@ -1306,25 +1321,20 @@ func (oc *Controller) deleteEgressNode(egressNode *kapi.Node) error {
 	if err != nil {
 		return fmt.Errorf("unable to list EgressIPs, err: %v", err)
 	}
-	for _, eIP := range egressIPs.Items {
-		keepStatusItems := []egressipv1.EgressIPStatusItem{}
-		for _, status := range eIP.Status.Items {
-			if status.Node != egressNode.Name {
-				// Keep only assignments that are not assigned to this node. Any
-				// egress IPs which are missing an assignment will try to be
-				// assigned once we've updated the object in reconcileEgressIP
-				keepStatusItems = append(keepStatusItems, status)
-			}
-		}
-		if len(keepStatusItems) != len(eIP.Status.Items) {
-			// Update the egress IP. The reconciliation loop for WatchEgressIP
-			// will try to assign the un-assigned IPs to whatever nodes remain.
-			// Perform a real update here. If we would call the reconciliation
-			// loop's function directly we'll need to update the object in that
-			// function and have the update recursively call the reconciliation
-			// loop again, i.e: perform two reconciliations for the same change.
-			if err := oc.updateEgressIPStatus(eIP.Name, keepStatusItems); err != nil {
-				klog.Errorf("Re-assignment for EgressIP: %s failed, unable to update object, err: %v", eIP.Name, err)
+	for _, egressIP := range egressIPs.Items {
+		for _, status := range egressIP.Status.Items {
+			if status.Node == egressNode.Name {
+				// Send a "synthetic update" on all egress IPs which have an
+				// assignment to this node. The reconciliation loop for
+				// WatchEgressIP will see that the current assignment status to
+				// this node is invalid and try to re-assign elsewhere. The
+				// workqueue's delta FIFO implementation will not trigger a
+				// watch event for updates on objects which have no semantic
+				// difference, hence: call the reconciliation function directly.
+				if err := oc.reconcileEgressIP(nil, &egressIP); err != nil {
```

Comment:
Yes, because in `validateEgressIPStatus` we validate if the status on the `new` object is valid, and before executing `deleteEgressNode` we mark the node as not usable for egress IP assignment, meaning `validateEgressIPStatus` will see that one current status assignment is invalid and we will try to re-assign that status elsewhere.